result.ts

```diff
@@ -0,0 +1,15 @@
+import type { Result } from "./types.ts";
+
+export function Ok<T>(value: T): Result<T> {
+  return {
+    ok: true,
+    value,
+  };
+}
+
+export function Err(error: Error): Result<never> {
```

Comment:
Will this work for all `T` in `Result<T>`? I'm not super familiar with how `never` behaves. We can also parameterize this:

```ts
export function Err<T>(error: Error): Result<T> {
  return { ok: true, value };
}
```

Even though the type-parameter isn't used in the return value, It will still present itself to the type checker as whatever concrete `Result<T>` is needing to be filled.

Finally,

What do you think about making these `const` functions? E.g.

```ts
export const Ok = <T>(value: T) => ({ ok: true, value });
```